packages/core/src/browser/tree/tree.ts

```diff
@@ -35,11 +35,11 @@ export interface ITree extends Disposable {
     /**
      * Refresh children of the root node.
      */
-    refresh(): void;
+    refresh(): Promise<boolean>;
     /**
      * Refresh children of the given node if it is valid.
      */
-    refresh(parent: Readonly<ICompositeTreeNode>): void;
+    refresh(parent: Readonly<ICompositeTreeNode>): Promise<boolean>;
```

Comment:
Why should it return boolean?

Reply:
Theoretically, node could not be `ICompositeTreeNode`, so refresh won't happen. The result is needed in expand node, when we wait for actual expanding which is done via refresh.

Reply:
It cannot happen, only composite tree nodes could be passed